controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

arm_controller:
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_pitch_joint
      - wrist_roll_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.6
      shoulder_pan_joint:
        trajectory: 0.1
        goal: 0.1
      shoulder_lift_joint:
        trajectory: 0.1
        goal: 0.1
      elbow_joint:
        trajectory: 0.1
        goal: 0.1
      wrist_pitch_joint:
        trajectory: 0.1
        goal: 0.1
      wrist_roll_joint:
        trajectory: 0.1
        goal: 0.1

gripper_controller:
  ros__parameters:
    joints:
      - gripper_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.6
      gripper_joint:
        trajectory: 0.1
        goal: 0.1
